plugins/Terminal/js/index.js

```diff
@@ -0,0 +1,224 @@
+	}
+
+	siac.on('error', () => {
+		console.log('\tPROGRAM ERRORED')
+		streamClosed()
+	})
+	siac.on('close', () => {
+		console.log('\tPROGRAM CLOSED')
+		streamClosed()
+	})
+
+	//If window is small auto close command overview so we can see the return value.
+	if (document.getElementsByClassName('command-history-list')[0].offsetHeight < 180) {
+		actions.hideCommandOverview()
+	}
+
+	return siac
+}
+
+
+const httpCommand = function(commandStr, actions) {
+	var commandString = commandStr
+	const originalCommand = commandStr.replace(/\s*\s/g, ' ').trim()
+
+	//Remove surrounding whitespace and leading siac command.
+	if (commandString.startsWith('siac')) {
+		commandString = commandString.slice(4).trim()
+	} else if (commandString.startsWith('./siac')) {
+		commandString = commandString.slice(6).trim()
+	}
+
+	//Parse arguments.
+	var args = commandString.split(' ')
+
+	//Add address flag to siac.
+	var siaAddr = url.parse('http://localhost:9980')
+
+	if (args.indexOf('-a') === -1 && args.indexOf('--address') === -1) {
+		if (SiaAPI.config.attr('address')) {
+			//Load default address.
+			siaAddr = url.parse('http://' + SiaAPI.config.attr('address'))
+		}
+	} else {
+		//Parse address flag.
+		var index = args.indexOf('-a')
+		if  (index === -1) {
+			index = args.indexOf('--address')
+
+		}
+		if (index < args.length-1) {
+			siaAddr = url.parse('http://' + args[index+1])
+		}
+		args.splice(index, 2)
+		commandString = args.join(' ')
+	}
+
+	var apiURL = ''
+	switch (commandString) {
+	case 'wallet unlock':
+		apiURL = '/wallet/unlock'
+		break
+
+	case 'wallet load seed':
+		apiURL = '/wallet/seed'
+		break
+
+	default:
+		console.log(`ERROR Command is not an http command. Rerouting. ${commandString}`)
+		return spawnCommand(commandString, actions).stdin
+	}
+
+	//Spawn new command if we are good to go.
+	var newCommand = Map({ command: originalCommand, result: '', id: Math.floor(Math.random()*1000000), stat: 'running' })
+	actions.addCommand(newCommand)
+
+	//Update the UI when the process receives new ouput.
+	var consumeChunk = function(chunk) {
+		console.log('Data chunk ' + chunk)
+		var newChunk = chunk
+		if (chunk.toString().trim() === '{"Success":true}') {
+			newChunk = 'Success'
+		}
+		actions.updateCommand(newCommand.get('command'), newCommand.get('id'), newChunk)
+	}
+
+	var closed = false
+	var streamClosed = function() {
+		if (!closed) {
+			actions.updateCommand(newCommand.get('command'), newCommand.get('id'), undefined, 'done')
+			closed = true
+		}
+	}
+
+	//If window is small auto close command overview so we can see the return value.
+	if (document.getElementsByClassName('command-history-list')[0].offsetHeight < 180) {
+		actions.hideCommandOverview()
+	}
+
+	var options = {
+		hostname: siaAddr.hostname,
+		port: siaAddr.port,
+		path: apiURL,
+		method: 'POST',
+		headers: {
+			'User-Agent': 'Sia-Agent',
+			'Content-Type': 'application/x-www-form-urlencoded',
+		},
+	}
+	var req = http.request(options, (res) => {
```

Comment:
Nice, that would simplify the code. My only concern is that would disallow the use of the `-a`/`--address` flag to be passed to these functions. Is that something we are ok with losing?